src/rgw/rgw_op.cc

```diff
@@ -369,6 +369,12 @@ int rgw_build_bucket_policies(RGWRados* store, struct req_state* s)
     if (!r) {
       if (!zonegroup.endpoints.empty()) {
 	s->zonegroup_endpoint = zonegroup.endpoints.front();
+      } else {
+        // use zonegroup's master zone endpoints
```

Comment:
@amitkumar50 Please check that the PR is targeting master before suggesting any changes.